packages/core/src/errors/InruptError.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @hidden
+ * Extends the regular JavaScript error object with extra meta-data (e.g.
+ * potentially data from a HTTP response, or with data from an RDF vocabulary
+ * (which may be imported locally, or looked up dynamically at runtime)).
+ * @packageDocumentation
+ */
+
+import { NamedNode } from "n3";
+
+import { VocabTerm } from "@inrupt/solid-common-vocab";
+
+export default class InruptError extends Error {
+  public readonly statusCode?: number;
+
+  public readonly statusText?: string;
+
+  constructor(
+    messageOrIri: string | NamedNode | VocabTerm,
+    messageParams?: string[],
```

Comment:
I'm not a fan of unnamed template parameters. I find it doesn't make the code easy to read, and doesn't provide context about how each parameter is going to be used. A `Record<string, string>` would be clearer, e.g.
```
const vocabError = UI_COMMON.errFileUpload_exceededSizeLimit;
const params = ["one", "two", "three"];
const error = new InruptError(vocabError, params);
```
would be easier to read as 
```
const vocabError = UI_COMMON.errFileUpload_exceededSizeLimit;
const params = {
  currentSize: "one",
  maxSize:  "two", 
  filename: "three"
};
const error = new InruptError(vocabError, params);
```

Reply:
Yep, totally agree - this is really the result of the original versions of this idea being written in Java, where constructing Maps is very verbose and clunky. Using positional params work quite well in any programming language, and a big advantage is that the JavaDoc/TSDoc/etc. generated by the Artifact Generator already includes the English message, thereby allowing developers (who use IDEs!) to see tooltip descriptions of the error message, from which determining the meaning and order of params is relatively straightforward (although I agree it's not as nice as fully named params). We discussed the idea of generating TypeScript types that could explicitly list the required param names, one type per error message, which could provide really nice IDE support, but would still leave the Java side in trouble I think.